### PR TITLE
refactor/analytics.coffee

### PR DESCRIPTION
**Pull request by sylvia0929**
_Any italic text should be deleted from the final Pull Request text, including this line_

# Description

This commit updates the analytics component and related AngularJS module to improve functionality and maintain compatibility with the latest requirements. The changes involve:

Migrating the analytics component to a more modern structure, enhancing performance and maintainability.
Refactoring the AngularJS module to ensure smoother integration with other components and the overall architecture.
Addressing any deprecated features or outdated practices in the existing analytics module to align with the project's current best practices.

Screenshot:

19411726832131_ pic
19421726832147_ pic

## Testing Checklist:

-Tested in latest Chrome
-Tested in latest Safari
-Tested in latest Firefox

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

**Comment by AusBruce** [comment]
I have test your code in my local machine, it runs well. before and after looks exactly the same. It is good to see you that you have make change in doubtfire.state.

**Comment by JackSCarroll** [comment]
These changes successfully unlink references to analytics.coffee and recreates the route in doubtfire.states to utilise the existing unit-analytics-route component files.
The old analytics.coffee and analytics.tpl.html files should be deleted now that they aren't being used.
There may be a need to rename this branch because this isn't a migration task.

File: src/app/config/config.coffee [hunk 1]
```diff
@@ -9,7 +9,7 @@ angular.module('doubtfire.config', [
   'doubtfire.config.vendor-dependencies'
   'doubtfire.config.local-storage'
   'doubtfire.config.routing'
-  'doubtfire.config.analytics'
+  #'doubtfire.config.analytics'
```
Comment (aNebula) [inline src/app/config/config.coffee:12]: - [ ] Remove commented lines from code

File: src/app/doubtfire-angularjs.module.ts [hunk 2]
```diff
@@ -60,7 +60,7 @@ import 'build/src/app/config/root-controller/root-controller.js';
 import 'build/src/app/config/local-storage/local-storage.js';
 import 'build/src/app/config/routing/routing.js';
 import 'build/src/app/config/vendor-dependencies/vendor-dependencies.js';
-import 'build/src/app/config/analytics/analytics.js';
+//import 'build/src/app/config/analytics/analytics.js';
```
Comment (aNebula) [inline src/app/doubtfire-angularjs.module.ts:63]: - [ ] Remove commented lines from code

File: src/app/doubtfire-angularjs.module.ts [hunk 3]
```diff
@@ -115,7 +115,7 @@ import 'build/src/app/units/states/rollover/directives/directives.js';
 import 'build/src/app/units/states/rollover/rollover.js';
 import 'build/src/app/units/states/index/index.js';
 import 'build/src/app/units/states/students-list/students-list.js';
-import 'build/src/app/units/states/analytics/analytics.js';
+//import 'build/src/app/units/states/analytics/analytics.js';
```
Comment (aNebula) [inline src/app/doubtfire-angularjs.module.ts:118]: - [ ] Remove commented lines from code

File: src/app/units/states/states.coffee [hunk 4]
```diff
@@ -5,7 +5,7 @@ angular.module('doubtfire.units.states', [
   'doubtfire.units.states.tasks'
   'doubtfire.units.states.groups'
   'doubtfire.units.states.students'
-  'doubtfire.units.states.analytics'
+  #'doubtfire.units.states.analytics'
```
Comment (aNebula) [inline src/app/units/states/states.coffee:8]: - [ ] Remove commented lines from code

**Review by aNebula — CHANGES_REQUESTED** [review]

**Comment by aNebula** [comment]
@sylvia0929 please remove the commented lines from code - comments should only be used to provide explanation or documentation of complex behaviour; comments should never be a replacement for deleting code.
After that, please open an upstream pull request against [9.x branch of doubtfire-lms/doubtfire-web](https://github.com/doubtfire-lms/doubtfire-web/tree/9.x) to add your changes to upstream and get credited for your contributions in your personal Github portfolio.

**Pull request closed without merging**